### PR TITLE
Release

### PR DESCRIPTION
**Pull request by ErwanAliasr1**
This PR is about adding the tooling to version & publish releases of ceph-nano.

Please check the following commits which are pretty well documented to get how it works.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 
 cn: clean
-	go build -i -v -ldflags="-X main.version=$(VERSION) -X main.tag=$(TAG) -X main.branch=$(BRANCH)"
+	go build -i -ldflags="-X main.version=$(VERSION) -X main.tag=$(TAG) -X main.branch=$(BRANCH)"
+	mv cn cn-$(TAG)-$(VERSION)
+	ln -sf "cn-$(TAG)-$(VERSION)" cn
 
 clean:
-	rm cn || true
+	rm cn &>/dev/null || true
```
Comment (leseb) [inline Makefile]: rm -f ?

Comment (leseb) [reply Makefile]: So we stay consistent

File: release.sh [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    fatal "Please define $variable_name"
+  fi
+}
+
+isGitRepositoryClean() {
+  git diff --no-ext-diff --quiet --exit-code
+}
+
+isGitTagExists() {
+    git tag -l | grep -qw "$1"
```
Comment (leseb) [inline release.sh]: Wrong indent?

File: release.sh [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+usage() {
+  cat << EOF
+  $0 - create a github release and upload files
+
+  -h            : show help message
+  -g <token>    : github token (can be defined with GITHUB_TOKEN variable)
+  -t <tag>      : tag to be released (can be defined with TAG variable)
+  -p <tag>      : previous tag to make the CHANGELOG (can be defined with PTAG variable)
+EOF
+exit 2
```
Comment (leseb) [inline release.sh]: Indent ?

File: release.sh [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+fatal() {
+  echo "$@"
+  if [ -e "$CHANGELOG" ]; then
+    rm -f $CHANGELOG
+  fi
+  exit 1
+}
+
+isBinaryExists() {
+  which $1 &>/dev/null|| fatal "Cannot find $1 binary, please check your environement"
```
Comment (leseb) [inline release.sh]: Null || 

(add space)

File: release.sh [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+isBinaryExists git
+
+isGitRepositoryClean || fatal "git repository is not clean, cannot make the release !"
+
+GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+GIT_LAST_COMMIT=$(git log --format="%H" -n 1)
+
+which github-release &>/dev/null
+if [ $? -ne 0 ]; then
+    echo "Installing github-release"
```
Comment (leseb) [inline release.sh]: Indent 2

File: release.sh [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    go get github.com/aktau/github-release
+    isBinaryExists github-release
+fi
+
+optspec=":hg:p:t:"
+while getopts "$optspec" optchar; do
+    case "${optchar}" in
```
Comment (leseb) [inline release.sh]: Indent 2

File: release.sh [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+  IMPLICIT_PTAG=$(git for-each-ref refs/tags --sort=-taggerdate --format='%(refname)' --count=1 | cut -d '/' -f 3)
+  if [ -z "$IMPLICIT_PTAG" ]; then
+    fatal "Cannot detect any previous release"
+  fi
+  while true; do
+    echo -n "Does $IMPLICIT_PTAG the git tag to consider for builiding the CHANGELOG ? (yes / no) "
+    read answer
+    # Testing lower case version of the answer
+    case ${answer,,} in
+        yes)
```
Comment (leseb) [inline release.sh]: Indent 2

**Review by leseb — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (ErwanAliasr1) [reply Makefile]: fixed

File: release.sh (hunk 4 above)
Comment (ErwanAliasr1) [reply release.sh]: fixed

File: release.sh (hunk 2 above)
Comment (ErwanAliasr1) [reply release.sh]: Fixed with a beautyfier

File: release.sh (hunk 3 above)
Comment (ErwanAliasr1) [reply release.sh]: Fixed with a beautyfier

File: release.sh (hunk 5 above)
Comment (ErwanAliasr1) [reply release.sh]: Fixed with a beautyfier

File: release.sh (hunk 6 above)
Comment (ErwanAliasr1) [reply release.sh]: Fixed with a beautyfier

File: release.sh (hunk 7 above)
Comment (ErwanAliasr1) [reply release.sh]: Fixed with a beautyfier

**Review by leseb — APPROVED** [review]

**Pull request merged**